src/roiextractors/multiimagingextractor.py

```diff
@@ -0,0 +1,107 @@
+from abc import ABC
+from array import ArrayType
+from typing import Tuple, List, Iterable
+
+import numpy as np
+
+from .extraction_tools import NumpyArray
+from .imagingextractor import ImagingExtractor
+
+
+class MultiImagingExtractor(ImagingExtractor, ABC):
+    """
+    This class is used to combine multiple ImagingExtractor objects by frame.
+    """
+
+    extractor_name = "MultiImagingExtractor"
+    installed = True
+    installation_mesg = ""
+
+    def __init__(self, imaging_extractors: List[ImagingExtractor]):
+        """
+        Parameters
+        ----------
+        imaging_extractors: list of ImagingExtractor
+            list of imaging extractor objects
+        """
+        super().__init__()
+        assert isinstance(imaging_extractors, list), "Enter a list of ImagingExtractor objects as argument"
+        assert all(isinstance(IX, ImagingExtractor) for IX in imaging_extractors)
+        self._imaging_extractors = imaging_extractors
+
+        # Checks that properties are consistent between extractors
+        self._check_consistency_between_imaging_extractors()
+
+        # Set properties based off the initial extractor
+        self._first_imaging_extractor = self._imaging_extractors[0]
+        self._num_channels = self._first_imaging_extractor.get_num_channels()
+        self._channel_names = self._first_imaging_extractor.get_channel_names()
+        self._sampling_frequency = self._first_imaging_extractor.get_sampling_frequency()
+        self._image_size = self._first_imaging_extractor.get_image_size()
+
+        self._start_frames, self._end_frames = [], []
+        num_frames = 0.0
+        for imaging_extractor in self._imaging_extractors:
+            self._start_frames.append(num_frames)
+            num_frames = num_frames + imaging_extractor.get_num_frames()
+            self._end_frames.append(num_frames)
+
+        self._num_frames = int(num_frames)
```

Comment:
Just curious, what was the reasoning behind starting the `num_frames` increment at `0.0` and then casting it back at the end? Can't you start it off as `int` and it will stay that way? (if it's overflow you're worried about, it would take an extremely large number of frames to have any effect on an `int32`, the Python default)